mayastor-test/e2e/node_disconnect/lib/node_disconnect_lib.go

```diff
@@ -0,0 +1,116 @@
+package node_disconnect_lib
+
+import (
+	"e2e-basic/common"
+	"fmt"
+	"os/exec"
+	"time"
+
+	. "github.com/onsi/gomega"
+)
+
+var (
+	defTimeoutSecs = "90s"
+)
+
+func DisconnectNode(vmname string, otherNodes []string, method string) {
+	for _, targetIP := range otherNodes {
+		cmd := exec.Command("bash", "../lib/io_connect_node.sh", vmname, targetIP, "DISCONNECT", method)
+		cmd.Dir = "./"
+		_, err := cmd.CombinedOutput()
+		Expect(err).ToNot(HaveOccurred())
+	}
+}
+
+func ReconnectNode(vmname string, otherNodes []string, checkError bool, method string) {
+	for _, targetIP := range otherNodes {
+		cmd := exec.Command("bash", "../lib/io_connect_node.sh", vmname, targetIP, "RECONNECT", method)
+		cmd.Dir = "./"
+		_, err := cmd.CombinedOutput()
+		if checkError {
+			Expect(err).ToNot(HaveOccurred())
+		}
+	}
+}
+
+// return the node name to isolate and a vector of IP addresses to isolate
+func GetNodes(uuid string) (string, []string) {
+	nodeList, err := common.GetNodeLocs()
+	Expect(err).ToNot(HaveOccurred())
+
+	var nodeToKill = ""
+	nexusNode := common.GetMsvNode(uuid)
+	fmt.Printf("nexus node is \"%s\"\n", nexusNode)
+
+	var otherAddresses []string
+
+	// find a node which is not the nexus
+	for _, node := range nodeList {
+		if node.NodeName != nexusNode && node.MayastorNode == true {
+			nodeToKill = node.NodeName
+			break
+		}
+	}
+	Expect(nodeToKill != "")
+
+	// get a list of the other ip addresses in the cluster
+	for _, node := range nodeList {
+		if node.NodeName != nodeToKill {
+			otherAddresses = append(otherAddresses, node.IPAddress)
+		}
+	}
+	Expect(len(otherAddresses) != 0)
+
+	fmt.Printf("node to kill is \"%s\"\n", nodeToKill)
+	return nodeToKill, otherAddresses
+}
+
+func LossTest(nodeToKill string, otherNodes []string, disconnectionMethod string, uuid string) {
```

Comment:
Nit - "nodeToKill" doesn't seem accurate as we are only disconnecting/reconnecting. Maybe "nodeToIsolate"?

Reply:
I was just thinking the same thing actually. A quick fix so I'll do it.

Reply:
Now pushed